src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -118,18 +103,18 @@ def update_metadata(self, metadata: Dict[str, Any]) -> None:
         snapshot's metadata"""
         self._metadata.update(_filter_nones(metadata))
 
-    def update_step(self, real_id: str, step: "Step") -> "PartialSnapshot":
-        step_update = _filter_nones(
-            {
-                "status": step.status,
-                "start_time": step.start_time,
-                "end_time": step.end_time,
-            }
+    def update_realization(
+        self,
+        real_id: str,
+        status: Optional[str] = None,
```

Comment:
Is this fine that it can be `None`?

Reply:
Hmm, that would be if you have an update for start- or end-time without having an update for the state. Not entirely sure if this is a scenario that actually occurs.

Reply:
You are right, it cannot happen that `status is None`. Will fix.

Reply:
Now changed. This implies that a `dict.get()` in `from_cloudevent` had to be changed to a direct accessor which can yield a `KeyError`. A buggy job_runner can then crash the evaluator by sending unknown events.

Reply:
test_dark_storage triggered this KeyError. Fixed by not mapping event type to state before we know what kind of event we are talking about.